cpp/Agents/EvolutionStrategy.hpp

```diff
@@ -0,0 +1,111 @@
+#include <iostream>
+#include <vector>
+#include <string>
+#include <mlpack/core.hpp>
+#include <mlpack/core/optimizers/sgd/sgd.hpp>
+#include "../environment.hpp"
+
+using namespace mlpack;
+using namespace std;
+using namespace arma;
+using namespace gym;
+
+class EvolutionStrategyAgent{
+ public:
+  mat model;
+  size_t frameW, frameH, num_actions;
+
+  EvolutionStrategyAgent(size_t frameW,
+                    size_t frameH,
+                    size_t num_actions)
+  :
+  frameW(frameW),
+  frameH(frameH),
+  num_actions(num_actions)
+  {
+    model = randn(num_actions, frameW * frameH * 3);
+  }
+
+  void Play(string environment,
+            string host,
+            string port)
+  {
+    size_t num_workers = 5;
+    double sigma = 1; // Rewards standard deviation
+    double alpha = 0.005; // Learning Rate
+    size_t input = frameW * frameH * 3;
+    
+    while(1)
+    {
+      mat workerRewards(num_workers, 1);
+      vector<mat> epsilons(num_workers);
+      vector<Environment> environments(num_workers,
+                          Environment(host, port, environment));
+
+      #pragma omp parallel for
+      for(size_t i = 0; i < num_workers; i++)
+      {
+        mat epsilon = randn(num_actions, input);
+        mat innerModel = model + (sigma * epsilon);
+
+        epsilons[i] = epsilon;
+
+        Environment& env = environments[i];
```

Comment:
You could also use ``boost::ptr_vector``:

```cpp
boost::ptr_vector<Environment> environments;
for (size_t i = 0; i < num_workers; ++i)
{
  environments.push_back(new Environment(host, port, environment));
}
```

in the parallel for loop you can use the index like:

```cpp
environments[0].compression(9);
```